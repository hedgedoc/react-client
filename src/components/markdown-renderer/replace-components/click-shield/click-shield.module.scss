/*
 * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
 *
 * SPDX-License-Identifier: AGPL-3.0-only
 */

.click-shield {

  position: relative;
  cursor: pointer;
  width: 100%;
  overflow: hidden;

  .preview-hover {
    color: white;
    opacity: 0.5;
    transition: opacity 0.2s;
    text-shadow: #000000 0 0 5px;

    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }

  .preview-hover-text {
    opacity: 0;
  }

  &:hover {
    .preview-hover-text {
      opacity: 1;
    }
    .preview-hover {
      opacity: 1;
      text-shadow: #000000 0 0 5px, #000000 0 0 10px;
    }
  }

  .preview-background {
    background: none;
    height: 100%;
    object-fit: cover;
    min-height: 300px;
    width: 100%;

    box-shadow: inset rgba(var(--bs-dark), 0.5) 0 0 20px;
  }
}
