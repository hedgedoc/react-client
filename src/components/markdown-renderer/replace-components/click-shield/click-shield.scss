/*
 * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
 *
 * SPDX-License-Identifier: AGPL-3.0-only
 */

.click-shield {

  position: relative;
  cursor: pointer;
  width: 100%;
  overflow: hidden;

  .preview-hover {
    display: inline;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    color: white;
    opacity: 0.5;
    transition: opacity 0.2s;
    text-shadow: #000000 0 0 5px;
  }

  .preview-hover-text {
    opacity: 0;
  }

  &:hover {
    .preview-hover-text {
      opacity: 1;
    }
    .preview-hover {
      opacity: 1;
      text-shadow: #000000 0 0 5px, #000000 0 0 10px;
    }
  }

  .preview-background {
    background: none;
    height: 100%;
    object-fit: cover;
    min-height: 300px;
    width: 100%;

    body.dark &{
      @import "../../../../style/variables.dark";
      box-shadow: inset fade-out($black, 0.5) 0 0 20px;
    }

    @import "../../../../style/variables.light";
    box-shadow: inset fade-out($black, 0.5) 0 0 20px;
  }
}
