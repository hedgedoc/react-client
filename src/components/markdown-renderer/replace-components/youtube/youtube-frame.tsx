/*
 SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)

 SPDX-License-Identifier: AGPL-3.0-only
 */

import React from 'react'
import { ClickShield } from '../click-shield/click-shield'
import type { IdProps } from '../custom-tag-with-id-component-replacer'

/**
 * Renders a video player embedding for https://youtube.com
 *
 * @param id The id from the youtube video url
 */
export const YouTubeFrame: React.FC<IdProps> = ({ id }) => {
  return (
    <ClickShield
      hoverIcon={'youtube-play'}
      targetDescription={'YouTube'}
      fallbackPreviewImageUrl={`https://i.ytimg.com/vi/${id}/maxresdefault.jpg`}
      fallbackBackgroundColor={'#ff0000'}
      data-cypress-id={'click-shield-youtube'}>
      <span className={'embed-responsive embed-responsive-16by9'}>
        <iframe
          className='embed-responsive-item'
          title={`youtube video of ${id}`}
          src={`https://www.youtube-nocookie.com/embed/${id}?autoplay=1`}
          allow='accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture'
        />
      </span>
    </ClickShield>
  )
}
