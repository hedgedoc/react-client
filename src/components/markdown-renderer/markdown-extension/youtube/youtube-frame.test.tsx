/*
 * SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
 *
 * SPDX-License-Identifier: AGPL-3.0-only
 */
import { render } from '@testing-library/react'
import { YouTubeFrame } from './youtube-frame'
import React from 'react'
import type { ClickShieldProps } from '../../replace-components/click-shield/click-shield'
import * as ClickShieldModule from '../../replace-components/click-shield/click-shield'

jest.mock('../../replace-components/click-shield/click-shield')

describe('YoutubeFrame', () => {
  beforeEach(() => {
    jest.spyOn(ClickShieldModule, 'ClickShield').mockImplementation((({ children }) => {
      return <span>This is a click shield for {children}</span>
    }) as React.FC<ClickShieldProps>)
  })

  it('renders a click shield', () => {
    const view = render(<YouTubeFrame id={'valid youtube id'} />)
    expect(view.container).toMatchSnapshot()
  })
})
