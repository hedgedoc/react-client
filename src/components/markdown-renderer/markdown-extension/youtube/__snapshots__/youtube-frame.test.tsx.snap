// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YoutubeFrame renders a click shield 1`] = `
<div>
  <span>
    This is a click shield for 
    <span
      class="embed-responsive embed-responsive-16by9"
    >
      <iframe
        allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
        class="embed-responsive-item"
        src="https://www.youtube-nocookie.com/embed/valid youtube id?autoplay=1"
        title="youtube video of valid youtube id"
      />
    </span>
  </span>
</div>
`;
